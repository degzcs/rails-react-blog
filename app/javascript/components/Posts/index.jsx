import React, { forwardRef } from "react";
import { Link } from "react-router-dom";
import BlogPost from "../../components/BlogPost/index.jsx";

const Loading = () => {
  return <p>Loading...</p>;
};

const Posts = ({ posts, hasNextPage, infiniteRef, rootRef }) => {
  return (
    <>
      <div className="py-5">
        <main className="container-fluid">
          <div className="row mt-5">
            <ul className="nav justify-content-end">
              <li className="nav-item">
                <Link className="btn btn-lg bg-light" to="blog/new-post/">
                  New Post
                </Link>
              </li>
            </ul>
          </div>
          <div className="row posts mb-5">
            {posts.map((post, index) => (
              <BlogPost key={index} post={post} />
            ))}
            {hasNextPage && (
              <div ref={infiniteRef}>
                <Loading />
              </div>
            )}
          </div>
        </main>
      </div>
    </>
  );
};

export default Posts;
