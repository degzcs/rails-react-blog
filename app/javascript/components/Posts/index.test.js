import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";

import Posts from "./index.jsx";

test("checks the rendering of the Posts", () => {
  const posts = [
    {
      title: "test",
      description: "test",
      slug: "test",
      image: "http://image.jpg",
    },
  ];

  const component = renderer.create(
    <StaticRouter location="someLocation">
      <Posts posts={posts} />
    </StaticRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
