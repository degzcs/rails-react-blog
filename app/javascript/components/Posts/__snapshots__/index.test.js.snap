// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the rendering of the Posts 1`] = `
<div
  className="py-5"
>
  <main
    className="container-fluid"
  >
    <div
      className="row mt-5"
    >
      <ul
        className="nav justify-content-end"
      >
        <li
          className="nav-item"
        >
          <a
            className="btn btn-lg bg-light"
            href="/blog/new-post/"
            onClick={[Function]}
          >
            New Post
          </a>
        </li>
      </ul>
    </div>
    <div
      className="row posts mb-5"
    >
      <a
        href="/blog/test"
        onClick={[Function]}
      >
        <div
          className="col-12 mt-3"
        >
          <div
            className="bcard"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-5  col-sm-12 card-body"
              >
                <h3
                  className="card-title"
                >
                  test
                </h3>
                <p
                  className=""
                >
                  test
                </p>
              </div>
              <div
                className="col-lg-7 col-sm-12 image-bk"
                style={
                  Object {
                    "backgroundImage": "url(http://image.jpg)",
                    "backgroundPosition": "50% 50%",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "100%",
                    "paddingBottom": "25%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </a>
    </div>
  </main>
</div>
`;
